File: зад2.swift
import Foundation
    var rings = ""
    print("Введите строку:  ")
    if let input = readLine() {
        rings = input
    }
//Проверка
 for i in 0...(rings.count-2) where i % 2 == 0  {
 let j = rings.substring(from: i).substring(to: 1)
 let j1 = rings.substring(from: i+1).substring(to: 1)
  if j=="R" || j=="B" ||  j=="W" {
   if j1 < "0" || j1>"9" {
    print("Значение было введено неверно")
   }
  }else {
   print("Значение было введено неверно")
  }
 }
  
var sticks: [Int] = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1] 
 //Считаем кол-во колец на каждом стержне
for i in 0...(rings.count-2) where i % 2 == 0 {
   let j = rings.substring(from: i).substring(to: 1)
   let j1 = rings.substring(from: i+1).substring(to: 1)
  if j == "R" {
   sticks[Int(j1)!] *= 2
  }else if j == "W" {
   sticks[Int(j1)!] *= 3
  }else if j == "B"{
   sticks[Int(j1)!] *= 5
  }
 }

 var counter = 0
 //На скольких кольцах стержни одного цвета
 for i in 0...(sticks.count-1) {
  if sticks[i] % 30 == 0 {
   counter += 1
  }
 }
 print(counter)

File: зад2.py
import math
rings = input("Введите строку:  ")
	#Проверка
for i in range(0, len(rings), 2): 
	if rings[i]=='R' or rings[i]=='B' or rings[i]=='W': 
		if rings[i + 1] < '0' or rings[i + 1]>'9':
			print("Значение было введено неверно")
			
	else:
		print("Значение было введено неверно")
		
sticks = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]	
	#Считаем кол-во колец на каждом стержне
for i in range(0, len(rings), 2):
	
	if rings[i] == 'R': 
		sticks[int(rings[i + 1])] *= 2
	elif rings[i] == 'W': 
		sticks[int(rings[i + 1])] *= 3
	elif rings[i] == 'B':
		sticks[int(rings[i + 1])] *= 5
counter = 0
	#На скольких кольцах стержни одного цвета
for i in sticks: 
	if i % 30 == 0 :
		counter+=1
print(counter)
